aci/resource_aci_interface_config.go

```diff
@@ -0,0 +1,420 @@
+
+		// To match the terraform plan stage changes with config file
+		if interfaceConfigMap["brkoutMap"] == "none" && d.Get("breakout").(string) == "" {
+			d.Set("breakout", nil)
+		} else {
+			d.Set("breakout", interfaceConfigMap["brkoutMap"])
+		}
+
+		d.Set("pc_member", interfaceConfigMap["pcMember"])
+	} else if portType == "fabric" {
+		interfaceConfig := models.FabricPortConfigurationFromContainer(interfaceConfigCont)
+		interfaceConfigMap, err = interfaceConfig.ToMap()
+	}
+
+	if err != nil {
+		return err
+	}
+
+	if interfaceConfigMap["dn"] == "" {
+		d.Set("breakout", nil)
+		d.Set("pc_member", nil)
+		return fmt.Errorf("Interface Configuration: %s not found", dn)
+	}
+
+	if nodeId, err := strconv.Atoi(interfaceConfigMap["node"]); err == nil {
+		d.Set("node", nodeId)
+	} else {
+		return err
+	}
+
+	d.Set("annotation", interfaceConfigMap["annotation"])
+	d.Set("interface", fmt.Sprintf("%s/%s/%s", interfaceConfigMap["card"], interfaceConfigMap["port"], interfaceConfigMap["subPort"]))
+	d.Set("role", interfaceConfigMap["role"])
+	d.Set("policy_group", interfaceConfigMap["assocGrp"])
+	d.Set("admin_state", getAdminState(interfaceConfigMap["shutdown"]))
+	d.Set("description", interfaceConfigMap["description"])
+	return nil
+}
+
+func resourceAciInterfaceConfigurationImport(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	log.Printf("[DEBUG] %s: Beginning Import", d.Id())
+	aciClient := m.(*client.Client)
+	dn := d.Id()
+
+	if match, _ := regexp.MatchString("/infra/", dn); match {
+		d.Set("port_type", "access")
+	} else if match, _ := regexp.MatchString("/fabric/", dn); match {
+		d.Set("port_type", "fabric")
+	} else {
+		return nil, fmt.Errorf("Interface Configuration DN: %s is not valid", dn)
+	}
+	err := getAndSetRemoteInterfaceConfiguration(dn, aciClient, d)
+	if err != nil {
+		return nil, err
+	}
+	log.Printf("[DEBUG] %s: Import finished successfully", d.Id())
+	return []*schema.ResourceData{d}, nil
+}
+
+func resourceAciInterfaceConfigurationCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	log.Printf("[DEBUG] Interface Configuration: Beginning Creation")
+	aciClient := m.(*client.Client)
+
+	node := strconv.Itoa(d.Get("node").(int))
+	interfaceAttrMap := make(map[string]string)
+	portType := d.Get("port_type").(string)
+	parsedInterface := strings.Split(getInterfaceVal(d.Get("interface")), "/")
+
+	if Annotation, ok := d.GetOk("annotation"); ok {
+		interfaceAttrMap["Annotation"] = Annotation.(string)
+	} else {
+		interfaceAttrMap["Annotation"] = "{}"
+	}
+
+	interfaceAttrMap["Card"] = parsedInterface[0]
+	interfaceAttrMap["Descr"] = d.Get("description").(string)
+	interfaceAttrMap["Node"] = node
+	interfaceAttrMap["Port"] = parsedInterface[1]
+	interfaceAttrMap["Role"] = d.Get("role").(string)
+	interfaceAttrMap["Shutdown"] = getAdminState(d.Get("admin_state").(string))
+	interfaceAttrMap["SubPort"] = parsedInterface[2]
```

Comment:
How does this work when 1/1 is passed?

Reply:
parsedInterface contains 3 elements, if there is no subPort then parsedInterface third index value will be 0.